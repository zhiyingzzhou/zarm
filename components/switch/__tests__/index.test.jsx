import React from 'react';
import { render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Switch from '../index';

describe('Switch', () => {
  it('renders correctly', () => {
    const wrapper = render(<Switch />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('defaultChecked', () => {
    const wrapper = render(<Switch defaultChecked />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('onChange', () => {
    const onChange = jest.fn();
    const wrapper = shallow(<Switch checked onChange={onChange} />);
    wrapper.find('input').simulate('change', { target: { checked: false } });
    expect(onChange).toBeCalledWith(false);
  });

  it('receive new checked', () => {
    const wrapper = shallow(<Switch />);
    wrapper.setProps({ disabled: true });
    wrapper.setProps({ checked: true });
  });

  it('receive new checked when disabled', () => {
    const onChange = jest.fn();
    const wrapper = shallow(<Switch disabled onChange={onChange} />);
    wrapper.find('input').simulate('change', { target: { checked: true } });
    expect(onChange).not.toBeCalled();
  });

  // it('size', () => {
  //   const wrapper = render(<Switch size="small" />);
  //   expect(toJson(wrapper)).toMatchSnapshot();
  // });
});
