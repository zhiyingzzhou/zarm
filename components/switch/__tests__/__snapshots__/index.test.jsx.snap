// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch defaultChecked 1`] = `
<span
  class="za-switch checked"
>
  <input
    checked=""
    class="za-switch-input"
    type="checkbox"
  />
</span>
`;

exports[`Switch renders correctly 1`] = `
<span
  class="za-switch"
>
  <input
    class="za-switch-input"
    type="checkbox"
  />
</span>
`;
